### PR TITLE
[Plugin] refine the httpsreq plugin implementation

### PR DESCRIPTION
**Pull request by 2019zhou**
https://github.com/WasmEdge/WasmEdge/issues/1763
https://github.com/WasmEdge/WasmEdge/issues/1768
https://github.com/WasmEdge/WasmEdge/issues/1788

### CODE REVIEW

**Comment by hydai** [comment]
Just a tip: Use `git-clang-format` to fix the format issue automatically. :-)

**Review by hydai — CHANGES_REQUESTED** [review]

File: plugins/wasmedge_httpsreq/httpsreqfunc.cpp [hunk 1]
```diff
@@ -15,62 +15,57 @@
...
-  std::copy_n(Host, HostLen, std::back_inserter(NewHost));
-  Env.Host = std::move(NewHost);
-
-  Env.Port = Port;
-
-  char *BodyStr = MemInst->getPointer<char *>(BodyPtr);
-  std::string NewBodyStr;
-  std::copy_n(BodyStr, BodyLen, std::back_inserter(NewBodyStr));
-  Env.BodyStr = std::move(NewBodyStr);
+  const char *Host = MemInst->getPointer<char *>(HostPtr);
```
Comment (hydai) [inline plugins/wasmedge_httpsreq/httpsreqfunc.cpp:30]: `<char *>` -> `<const char *>`

File: plugins/wasmedge_httpsreq/httpsreqfunc.cpp [hunk 2]
```diff
@@ -15,62 +15,57 @@
...
+  const char *Body = MemInst->getPointer<char *>(BodyPtr);
```
Comment (hydai) [inline plugins/wasmedge_httpsreq/httpsreqfunc.cpp:31]: `<char *>` -> `<const char *>`

File: plugins/wasmedge_httpsreq/httpsreqfunc.cpp [hunk 3]
```diff
@@ -90,9 +84,8 @@ Expect<void> SendData::body(const Runtime::CallingFrame &Frame,
   freeaddrinfo(Addrs);
 
   if (Sfd == -1) {
-    fprintf(stderr, "[Httpsreq plugin] %s: %s\n", Env.Host.c_str(),
-            strerror(Err));
-    abort();
+    spdlog::error("{}", strerror(Err));
```
Comment (hydai) [inline plugins/wasmedge_httpsreq/httpsreqfunc.cpp:87]: The output format is not the same as the previous ones.

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by 2019zhou** [comment]
done

File: plugins/wasmedge_httpsreq/httpsreqfunc.cpp [hunk 4]
```diff
@@ -15,62 +15,57 @@
...
 
-  std::sprintf(PortStr, "%d", Port);
-
-  Err = getaddrinfo(Env.Host.c_str(), PortStr, &Hints, &Addrs);
+  Err = getaddrinfo(HostStr.c_str(), PortStr.c_str(), &Hints, &Addrs);
   if (Err != 0) {
-    fprintf(stderr, "[Httpsreq plugin] %s: %s\n", Env.Host.c_str(),
-            gai_strerror(Err));
-    abort();
+    spdlog::error("[WasmEdge Httpsreq] {}", gai_strerror(Err));
```
Comment (hydai) [inline plugins/wasmedge_httpsreq/httpsreqfunc.cpp:67]: You remind me that we should use `WasmEdge` not `Wasmedge` in other places. Please check it again.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]
LGTM. @q82419 PTAL

**Review by q82419 — APPROVED** [review]

**Comment by hydai** [comment]
@q82419 Since most issues are fixed, should we add the `httpsreq` to the pre-built plugin list?

**Pull request merged**